ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/feemarket/LegacyFeeMarket.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet.feemarket;
+
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.core.fees.TransactionPriceCalculator;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class LegacyFeeMarket implements FeeMarket {
+
+  private final TransactionPriceCalculator txPriceCalculator;
+
+  public LegacyFeeMarket() {
+    this.txPriceCalculator = TransactionPriceCalculator.frontier();
+  }
+
+  @Override
+  public boolean implementsBaseFee() {
+    return false;
+  }
```

Comment:
Not saying we should do these but other options to consider:
* make this method `default` in `FeeMarket` and checks if the instance is `instanceof BaseFeeMarket`
* the calling context checks directly if the `FeeMarket` is `instanceof BaseFeeMarket` (this prevents these boolean methods from multiplying`

Reply:
yeah good idea.  I am not a fan of instanceof, but default and an override default is pretty clean :+1: